n2survey/lime/survey.py

```diff
@@ -280,6 +291,203 @@ def get_responses(
+                    condition_str = inner_logic.join(
+                        [f"{question} == '{answer}'" for answer in answers]
+                    )
+
+                # For multiple-choice questions, texts are converted to subquestion
+                # id using the choices dict
+                elif question_type == "multiple-choice":
+                    inverted_answer_dict = {
+                        answer: id for id, answer in self.get_choices(question).items()
+                    }
+                    subquestions = [inverted_answer_dict[answer] for answer in answers]
+                    condition_str = inner_logic.join(
+                        [f"{subquestion} == 'Y'" for subquestion in subquestions]
+                    )
+
+                # For other question types, e.g. "free", texts are left unchanged
+                else:
+                    condition_str = inner_logic.join(
+                        [f"{question} == {answer}" for answer in answers]
+                    )
+
+            # If only a single answer to question
+            else:
+                if question_type in ["single-choice", "array"]:
+                    inverted_answer_dict = {
+                        answer: id for id, answer in self.get_choices(question).items()
+                    }
+                    answers = inverted_answer_dict[answers]
+                    condition_str = f"{question} == '{answers}'"
+                elif question_type == "multiple-choice":
+                    inverted_answer_dict = {
+                        answer: id for id, answer in self.get_choices(question).items()
+                    }
+                    subquestion = inverted_answer_dict[answers]
+                    condition_str = f"{subquestion} == 'Y'"
+                else:
+                    condition_str = f"{question} == {answers}"
+            constructed_str = condition_str
+
+        return constructed_str
+
+    def keep_choices(self, conditions: Union[tuple, list]) -> "LimeSurvey":
+        """Filter responses DataFrame by keeping certain choices to specified questions
+
+        Args:
+            conditions (tuple or list of tuples): Conditions to re-format. Tuples in
+                format of (question_id, choices). choices can be a single str or list
+                of str. Multiple conditions are input as a list of tuples.
+                E.g. ("A6", "Woman") or [("A6", ["Woman", "Man"]), ("B6_SQ001", "Yes")]
+
+        Returns:
+            LimeSurvey: LimeSurvey with filtered responses
+        """
+
+        # Construct filter conditions str
+        constructed_filter_conditions = self.construct_filter_conditions(conditions)
+        # Make copy of LimeSurvey instance
+        filtered_survey = self._copy()
+        # Filter responses DataFrame
+        filtered_survey.responses = self.responses.query(constructed_filter_conditions)
+
+        return filtered_survey
```

Comment:
As far as I remember, we decided to give only one condition in the method call. And then chain the method calls if one needs more than one question condition, like `survey.keep_choices("A6", "Woman").keep_choices("B6_SQ001", "Yes")`, is it?

With this, it looks like the whole function (including construct_filter_conditions) can be reduced like:

```python
def keep_choices(self, question: str, choices: Union[str, list, tuple]) -> "LimeSurvey":
    choices = list(choices)
    return self.filter(
        # If choices are provided in codes
        self.get_responses(question, labels=False).isin(choices).any(axis='columns') |
        # If choices are provided in labels
        self.get_responses(question, labels=True).isin(choices).any(axis='columns')
    )
```

**Note** This is just an idea that has to be completed:
1) This would not cover "multiple-choice" questions. However, a similar approach can be used.
2) I used only `any(...)`, however, we might want to add a function argument that will check whether one wants `any` or `all` of sub-columns to be True